pseudocode-defns/pseudocode_defns.tex

```diff
@@ -0,0 +1,490 @@
+
+\lstdefinestyle{mystyle}{
+    backgroundcolor=\color{backcolour},   
+    commentstyle=\color{codegreen},
+    keywordstyle=\color{magenta},
+    numberstyle=\tiny\color{codegray},
+    stringstyle=\color{codepurple},
+    basicstyle=\ttfamily\footnotesize,
+    breakatwhitespace=false,         
+    breaklines=true,                 
+    captionpos=b,                    
+    keepspaces=true,                 
+    numbers=left,                    
+    numbersep=5pt,                  
+    showspaces=false,                
+    showstringspaces=false,
+    showtabs=false,                  
+    tabsize=2
+}
+
+\lstset{style=mystyle}
+
+\newcommand{\MultiSets}{\mathrm{MultiSets}}
+\newcommand{\len}{\mathrm{len}}
+\newcommand{\din}{d_{in}}
+\newcommand{\dout}{d_{out}}
+\newcommand{\Relation}{\mathrm{Relation}}
+\newcommand{\X}{\mathcal{X}}
+\newcommand{\Y}{\mathcal{Y}}
+\newcommand{\True}{\texttt{True}}
+\newcommand{\False}{\texttt{False}}
+\newcommand{\clamp}{\texttt{clamp}}
+\newcommand{\horizline}{\noindent\rule{\textwidth}{1pt}}
+
+
+\newcommand{\silvia}[1]{{ {\color{blue}{(silvia)~#1}}}}
+\newcommand{\grace}[1]{{ {\color{purple}{(grace)~#1}}}}
+\newcommand{\connor}[1]{{ {\color{teal}{(connor)~#1}}}}
+\newcommand{\sfi}{\text{\textcolor{olive}{(Salil's feedback incorporated.)}}}
+\newcommand{\todosfi}{\todo{Add Salil's feedback.}}
+
+\newcommand{\metricDefn}[1]{The definition of \emph{#1} in the ``proof definitions'' document tells how the distance between data is calculated.}
+
+\newcommand{\inRust}[2]{See \url{#2}.}
+\newcommand{\inOpenDPRust}[2]{In OpenDP (Rust), this is called \texttt{#1}. See \url{#2}.}
+\newcommand{\inPython}[2]{In Python, this is called \texttt{#1}. See \url{#2}.}
+\newcommand{\questionc}[1]{\textcolor{red}{\textbf{Question:} #1}}
+\newcommand{\questionr}[1]{\textcolor{blue}{\textbf{Question for reviewers:}}\textcolor{red}{~#1}}
+\newcommand{\T}{\texttt{T}}
+\newcommand{\data}{\texttt{data}}
+\newcommand{\todo}{{\textcolor{red}{TODO: }}}
+\newcommand{\todonei}{{\textcolor{red}{TODO (future -- not enough info yet): }}}
+\newcommand{\iffText}{\text{if and only if}}
+
+
+\title{List of definitions used in the pseudocode}
+\author{S\'ilvia Casacuberta, Grace Tian, and Connor Wagaman}
+\date{June--July 2021}
+
+\begin{document}
+
+\maketitle
+
+We use the following guideline: if a term appears in the preconditions \& pseudocode section, then this term is defined in the ``List of definitions used in the pseudocode" document. Otherwise, it appears in the ``List of definitions used in the proofs" document. 
+
+In both cases, we maintain the terms in alphabetical order within each section. ``TODOs'' should be included at the end of the corresponding section. On the other hand, ``TODOs'' which better specify an already-defined term should be included immediately following the definition of that term. Examples should never be part of the definition, but we encourage their use right after the definition of a term.
```

Comment:
I don't know that alphabetical order is the best - some definitions need to be read before other ones, and currently the dependency structure is not clear.   Maybe things should be done in a topological sort on dependencies, but there is also an index for looking up definitions alphabetically?

Reply:
I imagined the doc working more like a dictionary that people would refer to as they go through the pseudocode. If they come to a pseudocode thing they don’t know, then they’ll look at the corresponding definition, and if that definition relies on another definition, then they can look at the other definition, too.

We can restructure the document if needed, but I'm not sure (1) how difficult it would be to restructure (2) whether the restructuring would be useful or (3) whether/how often the structure of the document would need to be reconfigured as more definitions are added.

Reply:
Connor, that makes sense.  Then I think at some point we need to write an addition "Tutorial" document that gives a basic background on the framework and its implementation, which someone should read before getting started.

Reply:
One thing that would be helpful is to use latex cross-references for definition dependencies (when the definitions appear in the same document - else include a link to where the document can be found).